beacon.go

```diff
@@ -17,110 +17,164 @@
 	http.HandleFunc("/", handler)
 }
 
-func requestToSearch(r *http.Request) (*genomics.SearchVariantsRequest, string, error) {
-	allele := r.FormValue("allele")
-	search := &genomics.SearchVariantsRequest{
-		ReferenceName: r.FormValue("chromosome"),
-		VariantSetIds: config.variantSetIds,
+func handler(w http.ResponseWriter, r *http.Request) {
+	if err := validateServerConfig(); err != nil {
+		writeError(w, fmt.Errorf("validating server config: %v", err))
+		return
 	}
 
-	coord, err := strconv.ParseInt(r.FormValue("coordinate"), 10, 64)
+	refName, allele, coord, err := parseInput(r)
 	if err != nil {
-		return nil, "", err
+		writeError(w, err)
+		return
 	}
 
-	// Start is inclusive, End is exclusive.  Search exactly for coordinate.
-	search.Start = coord
-	search.End = search.Start + 1
+	ctx := appengine.NewContext(r)
+	exists, err := genomeExists(ctx, refName, allele, coord)
+	if err != nil {
+		writeError(w, err)
+		return
+	}
+
+	if err := writeResponse(w, exists); err != nil {
+		writeError(w, err)
+		return
+	}
+}
 
-	if search.ReferenceName == "" {
-		return nil, "", errors.New("missing reference name")
+func genomeExists(ctx context.Context, refName string, allele string, coord int64) (bool, error) {
+	// Start is inclusive, End is exclusive.  Search exactly for coordinate.
+	query := fmt.Sprintf(`
+		SELECT count(v.reference_name) as count
+		FROM %s as v
+		WHERE reference_name='%s'
+			AND v.start <= %d AND %d < v.end
+	 	 	AND reference_bases='%s'
+		LIMIT 1`,
+		fmt.Sprintf("`%s`", config.tableId),
+		refName,
+		coord,
+		coord+1,
+		allele)
+
+	bqclient, err := bigquery.NewClient(ctx, config.projectID)
+	if err != nil {
+		return false, fmt.Errorf("creating bigquery client: %v", err)
 	}
-	if allele == "" {
-		return nil, "", errors.New("missing allele")
+	it, err := bqclient.Query(query).Read(ctx)
+	if err != nil {
+		return false, fmt.Errorf("querying database: %v", err)
 	}
 
-	return search, allele, nil
+	var result struct {
+		Count int
+	}
+	if err := it.Next(&result); err != nil {
+		return false, fmt.Errorf("reading query result: %v", err)
+	}
+	return result.Count > 0, nil
 }
 
-func handler(w http.ResponseWriter, r *http.Request) {
-	c := appengine.NewContext(r)
-	search, allele, err := requestToSearch(r)
-	if err != nil {
-		http.Error(w, err.Error(), http.StatusBadRequest)
-		return
+func validateServerConfig() error {
+	if config.projectID == "" {
+		return fmt.Errorf("%s must be specified", projectKey)
+	}
+	if config.tableId == "" {
+		return fmt.Errorf("%s must be specified", bqTableKey)
 	}
+	return nil
+}
 
-	client, err := google.DefaultClient(c, genomics.GenomicsReadonlyScope)
-	if err != nil {
-		http.Error(w, "Invalid server configuration", http.StatusInternalServerError)
+func parseInput(r *http.Request) (string, string, int64, error) {
+	refName := r.FormValue("chromosome")
+	if refName == "" {
+		return "", "", 0, newBadRequestError("parsing chromosome name", errors.New("value is required"))
```

Comment:
Here, again, you should return bare errors and let the calling function annotate it with context:

{
  ...
   return "", "", 0, errors.New("missing chromosome name")
  ...
}

if err := parseInput(...); err != nil {
   return newBadRequestError("parsing input parameters", err)
}

Reply:
done